File: Server/Cookbook/src/main/java/ru/geekbrains/cookbook/repository/RecipeRepository.java
package ru.geekbrains.cookbook.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;
import ru.geekbrains.cookbook.domain.Recipe;

@Repository
public interface RecipeRepository extends JpaRepository<Recipe, Long>, JpaSpecificationExecutor<Recipe> {
}

File: app/src/main/java/ru/rumigor/cookbook/data/repository/RecipeRepository.kt
package ru.rumigor.cookbook.data.repository

import io.reactivex.rxjava3.core.Observable
import ru.rumigor.cookbook.data.model.Category
import ru.rumigor.cookbook.data.model.Recipe

interface RecipeRepository {
    fun getRecipes(): Observable<List<Recipe>>
    fun getRecipe(recipeID: String): Observable<Recipe>
    fun getCategories(): Observable<List<Category>>
//    fun addRecipe(recipe: Recipe): Observable<ResponseBody>
}